pkgs/development/libraries/fltk/1.4.nix

```
@@ -1,5 +1,185 @@
-import ./common.nix {
-  version = "1.4.x-2021-12-21";
-  rev = "c8bb2a35850be7c6eaec5ad5a2936a77f7913de2";
-  sha256 = "1fwfg1hp1ajqh2b4ra4phi96854q9i8c0gbyi7pr35ljyv848295";
-}
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  cmake,
+  pkg-config,
+  zlib,
+  libjpeg,
+  libpng,
+  fontconfig,
+  freetype,
+  libX11,
+  libXext,
+  libXinerama,
+  libXfixes,
+  libXcursor,
+  libXft,
+  libXrender,
+
+  withGL ? true,
+  libGL,
+  libGLU,
+  glew,
+
+  withCairo ? true,
+  cairo,
+
+  withPango ? stdenv.hostPlatform.isLinux,
+  pango,
+
+  withDocs ? true,
+  doxygen,
+  graphviz,
+
+  withExamples ? (stdenv.buildPlatform == stdenv.hostPlatform),
+}:
+
+let
+  onOff = value: if value then "ON" else "OFF";
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "fltk";
+  version = "1.4.4";
+
+  src = fetchFromGitHub {
+    owner = "fltk";
+    repo = "fltk";
+    rev = "release-${finalAttrs.version}";
+    hash = "sha256-xba9uiiwQhM4a3Rf4PoGshY3mUKZrcsm+Iv7qnE2sEA=";
+  };
+
+  outputs = [ "out" ] ++ lib.optional withExamples "bin" ++ lib.optional withDocs "doc";
+
+  # Manually move example & test binaries to $bin to avoid cyclic dependencies on dev binaries
+  outputBin = lib.optionalString withExamples "out";
+
+  postPatch = ''
+    patchShebangs documentation/make_*
+  '';
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ]
+  ++ lib.optionals withDocs [
+    doxygen
+    graphviz
+  ];
+
+  buildInputs =
+    lib.optionals (withGL && !stdenv.hostPlatform.isDarwin) [
+      libGL
+      libGLU
+    ]
+    ++ lib.optionals (withExamples && withGL) [
+      glew
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [
+      fontconfig
+    ];
+
+  propagatedBuildInputs = [
+    zlib
+    libjpeg
+    libpng
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isLinux [
+    freetype
+    libX11
+    libXext
+    libXinerama
+    libXfixes
+    libXcursor
+    libXft
+    libXrender
+  ]
+  ++ lib.optionals withCairo [
+    cairo
+  ]
+  ++ lib.optionals withPango [
+    pango
+  ];
+
+  cmakeFlags = [
+    # Common
+    "-DFLTK_BUILD_SHARED_LIBS=${onOff (!stdenv.hostPlatform.isStatic)}"
```

```suggestion
    (lib.cmakeBool "FLTK_BUILD_SHARED_LIBS" (!stdenv.hostPlatform.isStatic))
```

`cmakeBool` is preferable over some custom thing. Obviously applies to the other options too.